src/python/pants/task/testrunner_task_mixin.py

```diff
@@ -132,40 +132,33 @@ def __init__(self, xml_path, cause):
         self.xml_path = xml_path
         self.cause = cause
 
-    def parse_xml_file(path):
+    def parse_xml_file(xml_file_path):
       try:
-        xml = XmlParser.from_file(path)
-        for testcase in xml.parsed.getElementsByTagName('testcase'):
+        root = ET.parse(xml_file_path).getroot()
+        for testcase in root.iter('testcase'):
           test_info = {}
 
           try:
-            test_info.update({'time': float(testcase.getAttribute('time'))})
+            test_info.update({'time': float(testcase.attrib.get('time'))})
           except:
             test_info.update({'time': None})
 
           for attribute in testcase_attributes:
-            attribute_value = testcase.getAttribute(attribute)
-            if not attribute_value:
-              attribute_value = None
-            test_info[attribute] = attribute_value
-
-          test_error = testcase.getElementsByTagName('error')
-          test_fail = testcase.getElementsByTagName('failure')
-          test_skip = testcase.getElementsByTagName('skipped')
-
-          if test_fail:
-            test_info.update({'result_code': FAILURE})
-          elif test_error:
-            test_info.update({'result_code': ERROR})
-          elif test_skip:
-            test_info.update({'result_code': SKIPPED})
-          else:
-            test_info.update({'result_code': SUCCESS})
-
-          tests_in_path.update({testcase.getAttribute('name'): test_info})
-
-      except (XmlParser.XmlError, ValueError) as e:
-        error_handler(ParseError(path, e))
+            test_info[attribute] = testcase.attrib.get(attribute)
+
+          result = SUCCESS
+          for _ in testcase.iter('skipped'):
```

Comment:
I tried using `if next(testcase.iter('skipped'):` here instead but it didn't work.  Is there a nicer way to check if the iter has any elements?